### PR TITLE
Homework5: Krivosheev Konstantin

### PR DESCRIPTION
**Pull request by TexnoBY**
### PR checklist
Please make sure you've passed PR checklist
- [x] my PR passes CI checks (flake8 and pycodestyle)
- [x] my PR named according to guidelines.
- [x] my PR contains complete tasks descriptions for all tasks
- [x] my PR made from the fork feature branch to maxim homework# branch.
- [x] my PR contains only code that solves appropriate homework tasks. Redundant 
files (configs, etc. ) are put to .gitignore

### CODE REVIEW

File: hw2_eval.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+### Homework2 evaluation
```
Comment (belovmd) [inline hw2_eval.md]: redundant file in pr

File: src/homework2/task1.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+"""Напишите программу, которая считает общую цену.
+
+   Вводится M рублей и N копеек цена, а также количество S товара Посчитайте
+   общую цену в рублях и копейках за L товаров.
+"""
+
+
+def total_sum(m, n, s):
```
Comment (belovmd) [inline src/homework2/task1.py]: redundant files

File: src/homework5/task2.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    with open(my_path) as f:
+        for line in f:
+            temp = []
+            for num in line:
+                temp.append(num)
+            history.append(temp)
+
+    def wrapper(*args, **kwargs):
+        with open(my_path, 'a') as f:
+            print(*args, file=f)
```
Comment (belovmd) [inline src/homework5/task2.py]: it would be nice to add datetime and func name.

File: src/homework5/task3.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+    во вложенном while я проверяю, если сл число является последовательностью,
+
+    то я меняю границу. Затем записываю в ответ эти границы.
+
+    :param lst:
+    :return: str
+    """
+    lst = lst[::-1]
+    uns = ''
```
Comment (belovmd) [inline src/homework5/task3.py]: unclear naming

File: src/homework5/task3.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    while lst:
```
Comment (belovmd) [inline src/homework5/task3.py]: changes the initial list in outer scope

File: src/homework5/task4.py [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+    work_path = path.join(path.dirname(__file__), 'data', 'task4')
+    films = []
+    try:
+        with open(work_path + '\\ratings.list') as f:
+            for line in f:
+                try:
+                    items = line.split()
+                    # film line
+                    if not line.strip() or len(items[0]) != 10:
```
Comment (belovmd) [inline src/homework5/task4.py]: magic numbers

File: src/homework5/task4.py [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+                        continue
+                    votes = items[1]
+                    rate = items[2]
+                    name = []
+                    year = 0
+                    for item in items[3:]:
+                        is_year = item[1:-1].isdigit()
+                        if len(item) == 6 and is_year and not year:
```
Comment (belovmd) [inline src/homework5/task4.py]: magic number

File: src/homework5/task4.py [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+                try:
+                    items = line.split()
+                    # film line
+                    if not line.strip() or len(items[0]) != 10:
+                        continue
+                    votes = items[1]
+                    rate = items[2]
+                    name = []
+                    year = 0
+                    for item in items[3:]:
```
Comment (belovmd) [inline src/homework5/task4.py]: magic number

File: src/homework5/task4.py [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+    years.sort()
+    min_year = years[0]
+    max_year = years[-1]
+    dct_years = {}
+    while min_year <= max_year:
+        dct_years[min_year] = 0
+        min_year += 1
+    for year in years:
+        dct_years[year] += 1
+    with open(work_path + '\\years.txt', 'w') as f:
```
Comment (belovmd) [inline src/homework5/task4.py]: os.path.join()

File: tests/test_homework2.py [hunk 10]
```diff
@@ -0,0 +1,104 @@
+from string import punctuation
+
```
Comment (belovmd) [inline tests/test_homework2.py]: redundant file

**Review by belovmd — CHANGES_REQUESTED** [review]
contains files from hw2. 
please take a look at these issues

**Review by belovmd — COMMENTED** [review]
please address changes issues

File: src/homework5/task2.py (hunk 3 above)
Comment (TexnoBY) [reply src/homework5/task2.py]: add datetime and func name

**Review by TexnoBY — COMMENTED** [review]
apply all recommendation, refactor task4

**Review by belovmd — COMMENTED** [review]
9

**Pull request closed without merging**